hw26_paparazzi/js/paparazi.js

```diff
@@ -0,0 +1,22 @@
+function create(fn) {
+    let o = Object.create(fn.prototype);
+  
+    fn.call(o);
+  
+    return o;
+  }
+  
+  // ...
+  
+  structureUserInfo.prototype.f = function() {};
+  structureUserInfo.prototype.g = function() {};
+  
+  function structureUserInfo(role, name) {
+    this.role = role;
+    this.name = name;
+  }
```

Comment:
No, we don't need to create the function as a constructor. You have to implement the closure! Pay attention to hometask requirements. As you can see we should call the function like this `structureUserInfo('John')('Admin')`. It means `structureUserInfo` should return another function. Each function will have separate argument